src/middleware/AuthMiddleware.ts

```diff
@@ -0,0 +1,19 @@
+import { jwtDecode } from "jwt-decode";
+import { JwtToken, UserRole } from "../models/JwtToken";
+import express from "express";
+
+export const allowRoles = (allowedRoles: UserRole[]) => {
+    return (req: express.Request, res: express.Response, next: express.NextFunction) => {
+        if (!req.session.token) {
+            return res.status(401).send('Not logged in');
```

Comment:
Requirement is non-logged in users are redirected to login page

Reply:
done